### PR TITLE
DOC: Add example to rv_continuous.fit

### LINKED ISSUE #19225: stats.t.fit() with own optimizer (e.g. to use Nelder-Mead) fails due to "returning result too late"
### Describe your issue.

Using an own `optimizer` in e.g. `stats.t.fit`, the result "object" (`OptimizeResult`) is not returned but further passed to the optimizer as independent variable resulting in an `KeyError`. 

Specifically, the "object" `OptimizeResult`  is passed to the function `_unpack_loc_scale()` like the variables in previous optimization steps.  But `_unpack_loc_scale()`  takes tuples or lists but not final result objects (`OptimizeResult` ).

scipy_optim_error

Additionally, the starting value `x0` is not taken advantage of: 

scipy_optim_x0_not_respeted

### Reproducing Code Example

```python
import numpy as np
from scipy import stats as stats
from scipy.optimize import minimize

num_sim = 1000
degrees_freedom = 3
t_loc = 0
t_scale = 1

np.random.seed(42)
draws_t_dist = stats.distributions.t.rvs(degrees_freedom, t_loc, t_scale, size=num_sim)

# Nelder-Mead
def optimizer_NM(func, x0=3, args=(0, 1), disp=False):
    return minimize(func, x0, args, method="nelder-mead")


stats.t.fit(draws_t_dist, method="MLE", optimizer=optimizer_NM)
```

### Error message

```shell
`Thread stopped due to exception of type KeyError`
```

### SciPy/NumPy/Python version and system information

```shell
1.11.2 1.25.2 sys.version_info(major=3, minor=11, micro=1, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /usr/local/include
    lib directory: /usr/local/lib
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS=
      NO_LAPACK= NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP= HASWELL MAX_THREADS=2
    pc file directory: /usr/local/lib/pkgconfig
    version: 0.3.21.dev
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /usr/local/include
    lib directory: /usr/local/lib
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS=
      NO_LAPACK= NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP= HASWELL MAX_THREADS=2
    pc file directory: /usr/local/lib/pkgconfig
    version: 0.3.21.dev
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.10.4
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 0.29.36
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  pythran:
    include directory: ../../tmp/pip-build-env-tshsyzfz/overlay/lib/python3.11/site-packages/pythran
    version: 0.13.1
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /opt/python/cp311-cp311/bin/python
  version: '3.11'
```

### PR DESCRIPTION
**Pull request by m-maggi**
#### Reference issue
Closes gh-19225 

#### What does this implement/fix?
Extend the documentation of scipy.stats.rv_continuous.fit

### CODE REVIEW

File: scipy/stats/_distn_infrastructure.py [hunk 1]
```diff
@@ -2589,6 +2589,16 @@ def fit(self, data, *args, **kwds):
         >>> loc1, scale1 = norm.fit(x)
         >>> loc1, scale1
         (0.92087172783841631, 2.0015750750324668)
+
+        Using a custom optimizer:
+
+        >>> def optimizer_NM(func, x0=[0.6, 1.6], args=(), disp=0):
```
Comment (lucascolley) [inline scipy/stats/_distn_infrastructure.py:2595]: where did you take these argument values from / why did you choose these values?

**Review by lucascolley — COMMENTED** [review]
Looks good so far - could you also address this part of https://github.com/scipy/scipy/issues/19225#issuecomment-1714637380?

> It doesn't state what the optimizer should return or guarantee that user guesses will be respected.

File: scipy/stats/_distn_infrastructure.py (hunk 1 above)
Comment (m-maggi) [reply scipy/stats/_distn_infrastructure.py]: Thanks for reviewing. I replaced the normal example with the beta, as normal does not use the ``optimizer`` argument https://github.com/m-maggi/scipy/blob/f964b3c812eb911a18f30eb26c146108e020862e/scipy/stats/_continuous_distns.py#L393-L399

Furthermore I added 
> The optimizer must return the estimated parameters.

In the doctoring.

File: scipy/stats/_distn_infrastructure.py [hunk 2]
```diff
@@ -2559,12 +2560,26 @@ def fit(self, data, *args, **kwds):
...
 
         >>> a1, b1, loc1, scale1 = beta.fit(x)
+        >>> a1, b1, loc1, scale1
+        (0.998314523745146, 2.0351090001496326, 0.0004988689720895926, 1.0059326261917274)
+
+        The fit can be done also using a custom optimizer:
+
+        >>> from scipy.optimize import minimize
+        >>> def custom_optimizer(func, x0, args=(), disp=0):
+        ...     res = minimize(func, x0, args, method="COBYLA")
```
Comment (j-bowhay) [inline scipy/stats/_distn_infrastructure.py:2576]:
```suggestion
        ...     res = minimize(func, x0, args, method="COBYLA", disp=disp)
```

Comment (m-maggi) [reply scipy/stats/_distn_infrastructure.py:2576]: thanks for reviewing, I will actually remove the disp argument from the docstring as it is anyways set to 0 when calling fit in stats/_distn_infrastructure.py
https://github.com/scipy/scipy/blob/1004bb958b43426cd6cfd5cdf5baafa4f138902c/scipy/stats/_distn_infrastructure.py#L2640C57-L2640C57

```python3
vals = optimizer(func, x0, args=(data,), disp=0)
obj = func(vals, data)
```

File: scipy/stats/_distn_infrastructure.py [hunk 3]
```diff
@@ -2500,8 +2500,9 @@ def fit(self, data, *args, **kwds):
             - optimizer : The optimizer to use.  The optimizer must take
               ``func`` and starting position as the first two arguments,
               plus ``args`` (for extra arguments to pass to the
-              function to be optimized) and ``disp=0`` to suppress
-              output as keyword arguments.
+              function to be optimized) and ``disp``. 
+              The ``fit`` method calls always the optimizer with ``disp=0``.
```
Comment (mdhaber) [inline scipy/stats/_distn_infrastructure.py:2504]:
```suggestion
              The ``fit`` method always calls the optimizer with ``disp=0``.
```
or just 
```suggestion
              The ``fit`` method calls the optimizer with ``disp=0`` to suppress output.
```

File: scipy/stats/_distn_infrastructure.py [hunk 4]
```diff
@@ -2559,12 +2560,26 @@ def fit(self, data, *args, **kwds):
 
         >>> from scipy.stats import beta
         >>> a, b = 1., 2.
-        >>> x = beta.rvs(a, b, size=1000)
+        >>> x = beta.rvs(a, b, size=1000, random_state=80)
```
Comment (mdhaber) [inline scipy/stats/_distn_infrastructure.py:2563]: Elsewhere in the docs there has been a push to use `default_rng` with a larger seed to encourage use of `Generator`s and avoid repeated use of the same seed.

File: scipy/stats/_distn_infrastructure.py [hunk 5]
```diff
@@ -2559,12 +2560,26 @@ def fit(self, data, *args, **kwds):
...
 
         >>> a1, b1, loc1, scale1 = beta.fit(x)
+        >>> a1, b1, loc1, scale1
+        (0.998314523745146, 2.0351090001496326, 0.0004988689720895926, 1.0059326261917274)
+
+        The fit can be done also using a custom optimizer:
+
+        >>> from scipy.optimize import minimize
+        >>> def custom_optimizer(func, x0, args=(), disp=0):
+        ...     res = minimize(func, x0, args, method="COBYLA", options={"disp": disp})
```
Comment (mdhaber) [inline scipy/stats/_distn_infrastructure.py:2576]:
```suggestion
        ...     res = minimize(func, x0, args, method="slsqp", options={"disp": disp})
```
Let's avoid recommending use of COBYLA. SLSQP seems to be much more reliable.

**Review by mdhaber — COMMENTED** [review]
Thanks @m-maggi. A few comments.

**Review by mdhaber — APPROVED** [review]

**Pull request merged**